jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/VolumePermissionEnricher.java

```diff
@@ -178,6 +187,33 @@ private String getMountPath(List<Container> containers, String name){
                 throw new IllegalArgumentException("No matching volume mount found for volume "+ name);
             }
 
+            private ResourceRequirements createNewResourceRequirements() {
+                if (hasNoResourcesConfigured())
+                    return null;
+                ResourceRequirementsBuilder resourceRequirementsBuilder = new ResourceRequirementsBuilder();
+                resourceRequirementsBuilder.withLimits(createResourcesMap(Config.CPU_LIMIT, Config.MEMORY_LIMIT));
+                resourceRequirementsBuilder.withRequests(createResourcesMap(Config.CPU_REQUEST, Config.MEMORY_REQUEST));
+                return resourceRequirementsBuilder.build();
+            }
+
+            private Map<String, Quantity> createResourcesMap(Configs.Config cpu, Configs.Config memory) {
+                Map<String, Quantity> resourcesMap = new HashMap<>();
+                String cpuValue = getConfig(cpu);
+                if (StringUtils.isNotBlank(cpuValue)) {
+                    resourcesMap.put("cpu", new Quantity(cpuValue));
+                }
+                String memoryValue = getConfig(memory);
+                if (StringUtils.isNotBlank(memoryValue)) {
+                    resourcesMap.put("memory", new Quantity(memoryValue));
+                }
+                return resourcesMap;
+            }
+
+            private boolean hasNoResourcesConfigured() {
+                return createResourcesMap(Config.CPU_LIMIT, Config.MEMORY_LIMIT).isEmpty() &&
+                    createResourcesMap(Config.CPU_REQUEST, Config.MEMORY_LIMIT).isEmpty();
```

Comment:
Does it make any difference if we just create the `ResourceRequirements` with null values?

Also, is there a typo in the config properties used?

Reply:
yes, this looks like a typo. I've fixed it

Reply:
> Does it make any difference if we just create the ResourceRequirements with null values?

You forgot to reply about this one.

This is based on the fact that we're already adding null values to requests or limits in case one or the other is not specified in the config.

Reply:
I'm not sure if I understand your question correctly. If value is null, it won't be added to resource map.  We also have a check `hasNoResourcesConfigured()` in `createNewResourceRequirements()` method to return null if no requests/limits configuration is provided.

Reply:
Consider I only add a CPU_LIMIT, the `withRequests` method will be called with an empty map.
Or the opposite, if I only add a CPU_REQUEST, the `withLimits` method will be called with an empty map.
So considering these cases, is it worth checking `hasResourcesConfigured` in the `createNewResourceRequirements` method?

Reply:
I checked this, even if I change this method to only set when configuration is provided like this:

```java
private ResourceRequirements createNewResourceRequirements() {
    boolean limitsConfigured = hasResourcesConfigured(Config.CPU_LIMIT, Config.MEMORY_LIMIT);
    boolean requestsConfigured = hasResourcesConfigured(Config.CPU_REQUEST, Config.CPU_REQUEST);
    if (limitsConfigured || requestsConfigured) {
        ResourceRequirementsBuilder resourceRequirementsBuilder = new ResourceRequirementsBuilder();
        if (limitsConfigured) {
            resourceRequirementsBuilder.withLimits(createResourcesMap(Config.CPU_LIMIT, Config.MEMORY_LIMIT));
        }
        if (requestsConfigured) {
            resourceRequirementsBuilder.withRequests(createResourcesMap(Config.CPU_REQUEST, Config.MEMORY_REQUEST));
        }
        return resourceRequirementsBuilder.build();
    }
    return null;
}
```

I'm noticing that values which are not set via builder are initialized to empty maps

Reply:
Yes, this is also part of the changes introduced in Kubernetes Client 6

These are mainly the reasons about the check, which might be kind of redundant in the end.

Reply:
What shall we do here? Shall we leave it as it is?

Reply:
I'm basically telling you to remove the config check and leave the method as:
```java
            private ResourceRequirements createNewResourceRequirements() {
                return new ResourceRequirementsBuilder()
                  .withLimits(createResourcesMap(Config.CPU_LIMIT, Config.MEMORY_LIMIT))
                  .withRequests(createResourcesMap(Config.CPU_REQUEST, Config.MEMORY_REQUEST))
                  .build();
            }
```

Or even remove the `createNewResourceRequirements` altogether and initialize inline in the `createPvInitContainer` method.

Because, apparently it's harmless initializing the resource requirements with empty maps, right?